tests/deepsparse/pipelines/test_async_pipeline.py

```diff
@@ -40,7 +42,8 @@ def threaded_pipeline(executor):
     """
     Auto-del fixture for Threaded Pipeline
     """
-    yield Pipeline.create(task="question-answering", executor=executor)
+    with mock_engine(rng_seed=0):
+        yield Pipeline.create(task="question-answering", executor=executor)
```

Comment:
There's some weird interaction with the pytest.fixture decorator. When used together the unit tests below receive a the generator object instead of the pipeline.